src/utils/charconv.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+	Copyright (C) 2024
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#include "utils/charconv.hpp"
+#include <cctype>
+#include <cstdlib>
+#include <cassert>
+#include <string>
+
+#ifdef USE_FALLBACK_CHARCONV
+
+namespace
+{
+	template<typename T>
+	T string_to_floating_point(const char* str, char** str_end)
+	{
+		static_assert(sizeof(T) == 1, "Only specializations can be used");;
```

Comment:
This assert wouldn't trigger if T is char. Is that maybe bad? I don't see a char specialization.

Reply:
Probably not a big problem but if it compiles I'll change it to' == 0'